File: desing-pattens-1/src/modelo/implementacao/ICCC.java
package modelo.implementacao;

import interfaces.IImposto;
import modelo.Orcamento;

public class ICCC implements IImposto {

	@Override
	public double calcula(Orcamento orcamento) {

		double valor = orcamento.getValor();
		if (valor < 1000) {
			return valor * 0.05;
		} else if (valor <= 3000) {
			return valor * 0.07;
		}
		return valor * 0.08 + 30;
	}

}

File: desing-pattens-1/src/main/java/modelo/implementacao/imposto/ICCC.kt
package main.java.modelo.implementacao.imposto

import main.java.interfaces.IImposto
import main.java.modelo.Orcamento

class ICCC : IImposto {

    override fun calcula(orcamento: Orcamento): Double {

        val valor = orcamento.valor
        if (valor < 1000) {
            return valor * 0.05
        } else if (valor <= 3000) {
            return valor * 0.07
        }
        return valor * 0.08 + 30
    }

}
